src/generate-query.ts

```diff
@@ -509,9 +506,52 @@ function getMissingSlicingArg(
   if (usesSlicingArg) return null
 
   // Return the first slicing arguments:
```

Comment:
I'm having a bit of trouble understanding what the following is doing. Could you remind me? And while we're at it, we can improve this comment so it's more helpful next time.

Reply:
added better comment

Reply:
```
  // Returns the first slicing argument
  // if slicing argument can be split on '.'
  // then we have a nested slicing argument,
  // and we need to follow the path all the way
  // down to the leaf input field definition
```